### PR TITLE
Updated pyfits references to astropy.io.fits

### PR DESCRIPTION
**Pull request by RogueAstro**
Fix for issue #5062

### CODE REVIEW

File: astropy/io/fits/verify.py [hunk 1]
```diff
@@ -1,4 +1,4 @@
-# Licensed under a 3-clause BSD style license - see PYFITS.rst
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
```
Comment (pllim) [inline astropy/io/fits/verify.py]: Hmm, there is still `PYFITS.rst` license in the list of distributed licenses though.

Comment (saimn) [inline astropy/io/fits/verify.py]: This header is present at the top of almost all files in `astropy.io.fits`:
```
❯ grep -ir PYFITS.rst astropy/io/fits | wc -l
32
```
The license is the same (3 clause BSD), the copyright holder is not, not sure if it makes sense to keep the old license ?

Comment (pllim) [inline astropy/io/fits/verify.py]: Not sure what is the legally right thing to do when Astropy absorbs something, say, from STScI... Maybe Astropy Project Coordinators ( @taldcroft , @kelle , @astrofrog , and @eteq ) can clarify?

**Comment by pllim** [comment]
Note: Looks like someone else submitted an alternative at #5140. FYI.

**Comment by eteq** [comment]
I'm not sure what the answer is to whether or not the license should change from the pyfits license.  If it is changed, it should be changed *everywhere* it's mentioned, though.

Perhaps @perrygreenfield has some insight here?  The pyfits copyright holder is stated as being AURA, which in this context probably means STScI.  At this point it may indeed be reasonable to just drop the pyfits copyright completely, but we probably need at least some record that the copyright was transferred... (which I speculate could just be a message here?)

**Pull request closed without merging**